ContractToImplementorsSerializer.cs

```diff
@@ -0,0 +1,164 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Dolittle. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Dolittle.Assemblies;
+using Dolittle.Collections;
+using Dolittle.Logging;
+using Dolittle.Scheduling;
+
+namespace Dolittle.Types
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IContractToImplementorsSerializer"/>
+    /// </summary>
+    public class ContractToImplementorsSerializer : IContractToImplementorsSerializer
+    {
+        readonly ILogger _logger;
+        readonly ConcurrentDictionary<string, Type> _typesByName = new ConcurrentDictionary<string, Type>();
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ContractToImplementorsSerializer"/>
+        /// </summary>
+        /// <param name="scheduler"></param>
+        /// <param name="assemblies"></param>
+        /// <param name="logger"></param>
+        public ContractToImplementorsSerializer(
+            IScheduler scheduler,
+            IAssemblies assemblies,
+            ILogger logger)
+        {
+            _logger = logger;
+
+            var before = DateTime.UtcNow;
+
+            scheduler.PerformForEach(assemblies.GetAll(), assembly =>
+            {
+                try
+                {
+                    var types = assembly.GetTypes();
+                    foreach (var type in types)
+                    {
+                        if (!string.IsNullOrEmpty(type.AssemblyQualifiedName)) _typesByName[type.AssemblyQualifiedName] = type;
+                    }
+                }
+                catch (ReflectionTypeLoadException ex)
+                {
+                    foreach (var loaderException in ex.LoaderExceptions)
+                        _logger.Error($"Failed to load: {loaderException.Source} {loaderException.Message}");
+                }
+
+            });
+            Console.WriteLine($"Gathering Types : {DateTime.UtcNow.Subtract(before).ToString("G")}");
```

Comment:
Due to logging taking a significant amount of CPU, I needed to measure timing without logging on to get more accurate data. Since this is WIP - we can keep it. It is not used right now anyways.